### PR TITLE
Change DefaultResolver in XmlReaderSettings to XmlUrlResolver

### PR DESCRIPTION
**Pull request by sepidehMS**
Addresses a behavior difference between Core and Desktop. On Desktop, default resolver for XmlReaderSettings is `XmlUrlResolver`, whereas in Core it is `SystemPathResolver`.
This difference makes Core not allow string uri to be resolved in `XmlReader.Create` method, while Desktop does.

cc: @danmosemsft @krwq @tarekgh

### CODE REVIEW

File: src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs [hunk 1]
```diff
@@ -87,10 +87,34 @@ public static void TestResolveInvalidPaths()
...
-        TargetFrameworkMonikers.NetFramework)]    // Full framework uses XmlUrlResolver instead of SystemPathResolver (which doesn't allow access to this path)
+        TargetFrameworkMonikers.NetFramework)]    // Full framework uses WebRequest instead of HttpRequest
         public static void TestResolveInvalidPath()
         {
-            AssertInvalidPath("ftp://www.bing.com");
+            Assert.Throws<System.ArgumentException>(() => XmlReader.Create("ftp://www.bing.com")); // Only 'http' and 'https' schemes are allowed.
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)] // Full framework uses WebRequest instead of HttpRequest
```
Comment (danmosemsft) [inline src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs]: Please add the issue # for switching Core to WebRequest

Comment (sepidehMS) [inline src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs]: Right now, Desktop and Core throw different exceptions when a ftp uri is being passed in. That is because at the time of porting, WebRequest was not available and we had to use HttpRequest instead in [XmlDownloadManager](https://source.dot.net/#System.Private.Xml/System/Xml/XmlDownloadManager.cs,36). I will address that issue in a separate PR.

File: src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs [hunk 2]
```diff
@@ -87,10 +87,34 @@ public static void TestResolveInvalidPaths()
 
         [Fact]
         [SkipOnTargetFramework(TargetFrameworkMonikers.Uap |    //[ActiveIssue(13121)]  // Access to path is denied in UWP
-        TargetFrameworkMonikers.NetFramework)]    // Full framework uses XmlUrlResolver instead of SystemPathResolver (which doesn't allow access to this path)
+        TargetFrameworkMonikers.NetFramework)]    // Full framework uses WebRequest instead of HttpRequest
         public static void TestResolveInvalidPath()
         {
-            AssertInvalidPath("ftp://www.bing.com");
+            Assert.Throws<System.ArgumentException>(() => XmlReader.Create("ftp://www.bing.com")); // Only 'http' and 'https' schemes are allowed.
```
Comment (danmosemsft) [inline src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs]: Do we have existing tests for http and https? At least we should check we don't get an ArgumentException back.

File: src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs [hunk 3]
```diff
@@ -569,7 +569,7 @@ private void Initialize(XmlResolver resolver)
 
         private static XmlResolver CreateDefaultResolver()
         {
-            return new XmlSystemPathResolver();
+            return new XmlUrlResolver();
```
Comment (tarekgh) [inline src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs]: did you try to run the tests against UAP? nothing to worry more than just double checking

**Comment by tarekgh** [comment]
LGTM

File: src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs (hunk 2 above)
Comment (sepidehMS) [inline src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs]: I tested that locally and it gets resolved with no errors with this change. Is it OK to send a http request from our tests to some webpage?

Comment (tarekgh) [inline src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs]: You may look at the networking tests which should be doing something similar

Comment (sepidehMS) [inline src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs]: Correction: Looks like we actually do have tests for http and https:
- [ReadAsyncAfterInitializationWithUriThrows](https://github.com/dotnet/corefx/blob/master/src/System.Private.Xml/tests/XmlReader/Tests/AsyncReaderLateInitTests.cs#L72)
- [ReadAfterInitializationWithUriOnAsyncReaderTrows](https://github.com/dotnet/corefx/blob/master/src/System.Private.Xml/tests/XmlReader/Tests/AsyncReaderLateInitTests.cs#L83)

Comment (danmosemsft) [inline src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs]: Info on loopback server if you need it: https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/writing-tests.md

File: src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs (hunk 3 above)
Comment (sepidehMS) [inline src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs]: @tarekgh AFAIK UWP test runner on Helix does not allow resolving external paths and that's why these tests were disabled for UAP. Issue tracking this: https://github.com/dotnet/corefx/issues/13121

I did test this change on UAP locally and it threw the expected ArgumentException.

**Comment by sepidehMS** [comment]
Added new commit.

**Comment by sepidehMS** [comment]
@dotnet-bot test Innerloop OSX10.12 Release x64 Build and Test please

```
ManagedTests.DynamicCSharp.Conformance.dynamic.context.operate.regclass.regstrct029.regstrct029.Test.DynamicCSharpRunTest (from (empty))
Stacktrace

MESSAGE:
Assert.Equal() Failure\nExpected: 0\nActual: 2
+++++++++++++++++++
STACK TRACE:
```

File: src/System.Private.Xml/tests/XmlReader/Tests/AsyncReaderLateInitTests.cs [hunk 4]
```diff
@@ -68,23 +68,45 @@ public static void ReadAfterInitializationWithTextReaderOnAsyncReaderDoesNotThro
...
+            }
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)] // Full framework uses WebRequest instead of HttpRequest
+        public static void ReadAsyncAfterInitializationWithUriThrows_FullFramework()
+        {
+            using (XmlReader reader = XmlReader.Create("http://test.test/test.html", new XmlReaderSettings() { Async = true }))
+            {
+                Assert.Throws<System.Net.WebException>(() => reader.ReadAsync().GetAwaiter().GetResult());
```
Comment (stephentoub) [inline src/System.Private.Xml/tests/XmlReader/Tests/AsyncReaderLateInitTests.cs]: Are the differences better these two tests just the type of exception expected? I believe we have an Assert extensions helper for that.

Comment (sepidehMS) [inline src/System.Private.Xml/tests/XmlReader/Tests/AsyncReaderLateInitTests.cs]: @stephentoub Yes. I'm going to remove this difference and make Core match Desktop in another PR when this PR gets merged, so we can leave these as is. But out of curiousity, how can the extensions help here? Can you please give an example?

Comment (safern) [inline src/System.Private.Xml/tests/XmlReader/Tests/AsyncReaderLateInitTests.cs]: https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/AssertExtensions.cs#L63

Perhaps we should add an extension that doesn't expect a paramName
